sensor/kubernetes/certrefresh/tls_issuer_common.go

```diff
@@ -35,3 +49,167 @@ type certificateRefresherGetter func(certsDescription string, requestCertificate
+		certRefreshTimeout, i.certRefreshBackoff)
+
+	if refreshStartErr := i.certRefresher.Start(); refreshStartErr != nil {
+		return i.abortStart(errors.Wrap(refreshStartErr, "starting certificate refresher"))
+	}
+
+	log.Debugf("%s TLS issuer started.", i.componentName)
+	return nil
+}
+
+func (i *tlsIssuerImpl) abortStart(err error) error {
+	log.Errorf("%s TLS issuer start aborted due to error: %s", i.componentName, err)
+	i.Stop(err)
+	return err
+}
+
+func (i *tlsIssuerImpl) Stop(_ error) {
+	i.stopSig.Signal()
+	if i.certRefresher != nil {
+		i.certRefresher.Stop()
+		i.certRefresher = nil
+	}
+
+	log.Debugf("%s TLS issuer stopped.", i.componentName)
+}
+
+func (i *tlsIssuerImpl) Notify(common.SensorComponentEvent) {}
+
+func (i *tlsIssuerImpl) Capabilities() []centralsensor.SensorCapability {
+	return []centralsensor.SensorCapability{i.sensorCapability}
+}
+
+// ResponsesC is called "responses" because for other SensorComponent it is Central that
+// initiates the interaction. However, here it is Sensor which sends a request to Central.
+func (i *tlsIssuerImpl) ResponsesC() <-chan *message.ExpiringMessage {
+	return i.msgToCentralC
+}
+
+// ProcessMessage dispatches Central's messages to Sensor received via the Central receiver.
+// This method must not block as it would prevent centralReceiverImpl from sending messages
+// to other SensorComponents.
+func (i *tlsIssuerImpl) ProcessMessage(msg *central.MsgToSensor) error {
+	response := i.getResponseFn(msg)
+	if response == nil {
+		// messages not supported by this component are ignored
+		return nil
+	}
+
+	go i.dispatch(response)
+	return nil
+}
+
+func (i *tlsIssuerImpl) dispatch(response *Response) {
+	if !i.requestOngoing.Load() {
+		log.Warnf("Received a response for request ID %q, but there is no ongoing RequestCertificates call.", response.RequestId)
+		return
+	}
+	if i.ongoingRequestID != response.RequestId {
+		log.Warnf("Request ID %q does not match ongoing request ID %q.",
+			response.RequestId, i.ongoingRequestID)
+		return
+	}
+
+	i.responseFromCentral = response
+	i.responseReceived.Signal()
+}
+
+// requestCertificates makes a new request for a new set of secured cluster certificates from Central.
+// Concurrent requests are *not* supported.
+func (i *tlsIssuerImpl) requestCertificates(ctx context.Context) (*Response, error) {
+	if i.requiredCentralCapability != nil {
+		// Central capabilities are only available after this component is created,
+		// which is why this check is done here
+		if !centralcaps.Has(*i.requiredCentralCapability) {
+			return nil, fmt.Errorf("TLS certificate refresh failed: missing Central capability '%s'", *i.requiredCentralCapability)
+		}
+	}
+
+	if !i.requestOngoing.CompareAndSwap(false, true) {
+		return nil, errors.New("concurrent requests are not supported.")
+	}
+
+	defer func() {
+		i.requestOngoing.Store(false)
+		i.responseReceived.Reset()
+	}()
```

Comment:
Actually yes, I think it should be called also if `ctx` is cancelled, to reset the component so that  new `requestCertificate` calls can be made.  Otherwise `requestOngoing` will remain forever `true` and future calls to `requestCertificate` will fail.

If somehow a response arrives after the context expires here, `requestOngoing` will be `false` and `dispatch` will ignore the response.